app/models/account.rb

```diff
@@ -62,7 +62,7 @@ class Account < ApplicationRecord
   )
 
   USERNAME_RE   = /[a-z0-9_]+([a-z0-9_\.-]+[a-z0-9_]+)?/i
-  MENTION_RE    = /(?<=^|[^\/[:word:]])@((#{USERNAME_RE})(?:@[[:word:]\.\-]+[[:word:]]+)?)/i
+  MENTION_RE    = /(@(?:[a-z0-9_]+(?:#{USERNAME_RE})?)+?(?:@(?:[\w\.\-]+[\w]+)+?)?)/i
```

Comment:
`:word:` is:

> A character in one of the following Unicode general categories Letter, Mark, Number, Connector_Punctuation

You are replacing it with `a-z0-9_` (which is equivalent to `\w`). I suspect that this regex might not work as expected.

Reply:
the one i came up with is `(@(?:[a-z0-9_]+(?:[a-z0-9_\.-]+[a-z0-9_]+)?)+?(?:@(?:[\w\.\-]+[\w]+)+?)?)` but i wanted to include the USERNAME_RE in the definition -- i may have done that incorrectly

Reply:
hmm, testing in `irb` indicates that `[[:word]]` is actually equivalent to `[\w]` or `[a-z0-9_]`